From f645128082117a0152a95b3dccd869a184b7513f Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 10 Aug 2022 01:23:48 -0700
Subject: [PATCH] Use posix strerror_r with musl

Default with glibc is GNU extention of strerror_r
where as musl uses posix variant, call that out

Upstream-Status: Inappropriate [Need wider porting beyond linux/musl/glibc]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Resolved conflicts while upgrading recipe from v1.9.9 to v3.1.9.

Signed-off-by: Niko Mauno <niko.mauno@vaisala.com>

diff --git a/src/flb_network.c b/src/flb_network.c
index d183209fd..41e0281b5 100644
--- a/src/flb_network.c
+++ b/src/flb_network.c
@@ -553,7 +553,7 @@ static int net_connect_async(int fd,
             /* Connection is broken, not much to do here */
 #if ((defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L) ||    \
      (defined(_XOPEN_SOURCE) || _XOPEN_SOURCE - 0L >= 600L)) &&     \
-  (!defined(_GNU_SOURCE))
+  (!defined(_GNU_SOURCE)) || (!defined(__GLIBC__))
             ret = strerror_r(error, so_error_buf, sizeof(so_error_buf));
             if (ret == 0) {
                 str = so_error_buf;
